src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs

```diff
@@ -1,270 +1,101 @@
-                    if (shouldAddInflight != 0)
-                    {
-                        jsObject?.AddInFlight();
-                    }
-                    result = jsObject;
-                    return;
+                    JSHostImplementation.s_gcHandleFromJSOwnedObject.Remove(handle.Target!);
+                    handle.Free();
                 }
+                arg_exc.Initialize();
             }
-            result = null;
-        }
-
-        [MethodImplAttribute(MethodImplOptions.NoInlining)] // https://github.com/dotnet/runtime/issues/71425
-        public static IntPtr GetCSOwnedObjectJSHandleRef(in JSObject jsObject, int shouldAddInflight)
-        {
-            jsObject.AssertNotDisposed();
-
-            if (shouldAddInflight != 0)
+            catch (Exception ex)
             {
-                jsObject.AddInFlight();
+                arg_exc.ToJS(ex);
             }
-            return jsObject.JSHandle;
         }
 
         [MethodImplAttribute(MethodImplOptions.NoInlining)] // https://github.com/dotnet/runtime/issues/71425
-        public static IntPtr TryGetCSOwnedObjectJSHandleRef(in object rawObj, int shouldAddInflight)
+        public static void CreateTaskCallback(JSMarshalerArgument* arguments_buffer)
         {
-            JSObject? jsObject = rawObj as JSObject;
-            if (jsObject != null && shouldAddInflight != 0)
+            ref JSMarshalerArgument arg_exc = ref arguments_buffer[0];
+            ref JSMarshalerArgument arg_return = ref arguments_buffer[1];
+            try
             {
-                jsObject.AddInFlight();
+                JSHostImplementation.TaskCallback holder = new JSHostImplementation.TaskCallback();
+                arg_return.slot.Type = MarshalerType.Object;
+                arg_return.slot.GCHandle = JSHostImplementation.GetJSOwnedObjectGCHandle(holder);
+            }
+            catch (Exception ex)
+            {
+                arg_exc.ToJS(ex);
             }
-            return jsObject?.JSHandle ?? IntPtr.Zero;
         }
 
         [MethodImplAttribute(MethodImplOptions.NoInlining)] // https://github.com/dotnet/runtime/issues/71425
-        public static void CreateCSOwnedProxyRef(IntPtr jsHandle, JSHostImplementation.MappedType mappedType, int shouldAddInflight, out JSObject jsObject)
+        public static void CallDelegate(JSMarshalerArgument* arguments_buffer)
         {
-            JSObject? res = null;
-
-            lock (JSHostImplementation.s_csOwnedObjects)
+            ref JSMarshalerArgument arg_exc = ref arguments_buffer[0];
+            ref JSMarshalerArgument arg_return = ref arguments_buffer[1];
+            try
             {
-                if (!JSHostImplementation.s_csOwnedObjects.TryGetValue((int)jsHandle, out WeakReference<JSObject>? reference) ||
-                    !reference.TryGetTarget(out res) ||
-                    res.IsDisposed)
+                GCHandle callback_gc_handle = (GCHandle)arg_return.slot.GCHandle;
```

Comment:
are we repurposing the return slot to hold an inbound argument here? if so, please specify that in a comment, it's subtle

Reply:
I will change it to regular arg